CAIPs/caip-320.md

```diff
@@ -0,0 +1,149 @@
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written in
+uppercase in this document are to be interpreted as described in [RFC
+2119][]
+
+### Definition
+
+The JSON-RPC provider is able to invoke a single JSON-RPC request accompanied by a [CAIP-2][] compatible `chainId` authorized by a pre-existing session.
+If that pre-existing session was initiated by a [CAIP-25] response containing a [sessionId][CAIP-171], this `sessionId` value should also be returned at the top level of the `wallet_switchChain` envelope (see [CAIP-316] for more context on managing sessions with and without `sessionId` keys).
+
+### Request
+
+The application would interface with an JSON-RPC provider to make request as follows:
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_switchChain",
+  "params": {
+    "sessionId": "0xdeadbeef",
+    "chainId": "eip155:1"
+  }
+}
+```
+
+The JSON-RPC method is labeled as `wallet_switchChain` and expects two parameters, **two of them required**:
+
+- **sessionId** (conditional) - [CAIP-171][] `SessionId` disambiguates an open session in a multi-session actor; it is required in some sessions, such as [CAIP-25][] sessions created by a response containing one, and SHOULD be omitted in other sessions, such as [CAIP-25] sessions created by a response not containing one (see [CAIP-316]).
+- **chainId** (required) - a valid [CAIP-2][] network identifier, previously authorized by or within a `scopeObject` in the active session
+
+### Validation
+
+1. A respondent SHOULD check the `scope` against active session's `scopeObject`s before executing or responding to such a request, and SHOULD invalidate a request for a scope not previously authorized.
+2. The respondent SHOULD check that `request.method` is authorized for the specified scope, and SHOULD invalidate a request for a scope not previously authorized.
+3. The respondent MAY check that the `request.params` are valid for `request.method`, if its syntax is known to it.
+4. The respondent MAY apply other logic or validation.
+5. The respondent MAY chose to drop invalid requests or return an error message.
+
+### Response
+
+Upon successful validation, the respondent will switch the current active chain to the targeted network.
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "sessionId": "0xdeadbeef",
+    "chainId": "eip155:1"
+  }
+}
+```
+
+#### Error Handling
+
+##### Generic Failure Code
+
+Unless the dapp is known to the wallet and trusted, the generic/undefined error response,
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "error": {
+    "code": 0,
+    "message": "Unknown error"
+  }
+}
+```
+
+##### Trusted Failure Codes
+
+More informative error messages MAY be sent in trusted-counterparty circumstances.
+The core error messages over trusted connections are as follows:
+
+The valid error messages codes are the following:
```

Comment:
Once I'm clearer on the answers above, I _may_ have to return to this section, it's still a little unclear to me if I should be evaluating these error codes as a _subset_ of CAIP-25 errors applicable/valid for this method, or as an _alternate set_ of errors that would be applicable if this were called _outside_ of a CAIP-25 connection!